src/sdk/namespace_fs.js

```diff
@@ -1964,69 +1974,81 @@ class NamespaceFS {
+            ) {
+                dbg.log0(
+                    'namespace_fs.restore_object: state - (!restore_request, !restore_expiry)',
+                    'or',
+                    '(restore_request, restore_expiry <= now)'
+                );
+
+                // First add it to the log and then add the extended attribute as if we fail after
+                // this point then the restore request can be triggered again without issue but
+                // the reverse doesn't works.
+                await this.append_to_restore_wal(file_path);
+
+                await file.replacexattr(fs_context, {
+                    [GlacierBackend.XATTR_RESTORE_REQUEST]: params.days.toString(),
+                });
 
-                    // Should result in HTTP: 200 OK
-                    return false;
-                }
+                // Should result in HTTP: 202 Accepted
+                return true;
+            }
 
-                // If neither the restore is ongoing nor a restore expiry is set, we can set the xattr
-                // essentially adding the object to the batch.
-                if (!restore_expiry || restore_expiry <= now) {
-                    dbg.log0('namespace_fs.restore_object: state - (!restore_request, !restore_ongoing, !restore_expiry or restore_expiry <= now)');
+            // 1 state
+            if (!restore_request && restore_expiry > now) {
+                dbg.log0('namespace_fs.restore_object: state - (!restore_request, restore_expiry > now)');
 
-                    // First add it to the log and then add the extended attribute as if we fail after
-                    // this point then the restore request can be triggered again without issue but
-                    // the reverse doesn't works.
-                    await this.append_to_restore_wal(file_path);
+                const expires_on = now;
+                expires_on.setUTCDate(expires_on.getUTCDate() + params.days);
+                expires_on.setUTCHours(0, 0, 0, 0);
 
-                    await file.replacexattr(fs_context, {
-                        [GlacierBackend.XATTR_RESTORE_REQUEST]: params.days.toString(),
-                    });
+                await file.replacexattr(fs_context, {
+                    [GlacierBackend.XATTR_RESTORE_EXPIRY]: expires_on.toISOString(),
+                });
 
-                    // Should result in HTTP: 202 Accepted
-                    return true;
-                }
+                // Should result in HTTP: 200 OK
+                return false;
             }
 
-            // 3 Invalid States
-            // 1. Restore request is set along with 1 or more than 1 attribute (2 different states)
-            // 2. Restore request is unset but both request ongoing and expiry are set
+            // Should be unreachable
+            dbg.error('namespace_fs.restore_object: unexpected state - reached unreachable');
             throw new RpcError('INTERNAL_ERROR');
```

Comment:
Can we do something better other than error 500? This would just result a user complaint so if there is any fix we suggest lets add it to be automatic.

Reply:
I think for this case it is better that we do throw error. There should be just 6 combination possible, any 7th combination should not be possible and probably it is better that we get to take a look at the logs if and whenever we hit this rather than something silently keep failing and then it gets reported later on presenting itself as a totally different issue. WDYT?